DEX.md

```diff
@@ -0,0 +1,93 @@
+
+In the uniform liquidity model (used in Uniswap V2), liquidity is defined as:
+
+$L^2 = reserve_0 * reserve_1$
+
+This formula is known as *Constant Product Invariant*. When trading, the liquidity remains unchanged:
+
+$reserve_0 * reserve_1 = (reserve_0 + amount_0) * (reserve_1 - amount_1)$
+
+$amount_1$ is the amount of $T_1$ that is going to be removed from the pool. Before sending
+it to the user, some fees are deducted: mainly a platform fee that goes to the owners of the
+DEX and an LP-fee that is going to be accumulated for the liquidity providers proportional to the 
+amount of liquidity they have added to the system.
+
+# Withdrawing Liquidity
+
+Liquidity providers can withdraw their tokens at any time. However, due to trades happening in between, they rarely receive the same token amounts they initially deposited. Instead, they withdraw, depending on the net direction of trades:
+
+1. More of one token,
+2. Less of the other, and
+3. The fees they have accrued during trading activity.
+
+Some DEXes provide the LPs with the functionality to withdraw their accrued fee and, if desired, reinvest it to
+the protocol to compound their rewards.
+
+# Pitfalls and Fallacies
+
+In this section, we will highlight common pitfalls and misconceptions that can be confusing for users.
+
+## Pool Price $\ne$ Actual Price
+
+The pool exchange price (which is called *Local Price*) is solely defined by the ratio of assets in the pool. As users can trade their tokens, the ratio of the tokens diverges from the actual price.
+
+## Impermanent Loss
+
+When LPs deposit assets to a pool, the exchange price can change according to the point $MR$. As mentioned above, the local price can be different from the actual price. Hence, if one asset, e.g., $T_0$, is significantly more appreciated than the other one. In this case, liquidity providers end up selling their assets at an undervalued price relative to the broader market. 
+
+## Slippage
+
+Traders don’t swap at a single fixed price, but along the curve defined by the constant product rule. Larger trades move the reserves more, so the average execution price is worse than the quoted start price.
+
+## Front-running
+
+When used as an extension to SNS, each initialization, deposit, or withdraw proposal goes through a public voting process. It means that an attacker can guess the outcome of a voting when in its last stage. As explained before, they can then front-run the execution of the proposal by submitting a trade transaction and move the market ratio away from the ratio expected in the proposal.
+
+## Liquidity Provider Risks
+
+Although LPs are entitled to receive LP-fees, LP fees do not always offset impermanent loss. As a result, LPs may end up with fewer assets in value than if they had simply held their tokens.
+
+## Transaction Fees
+
+Every deposit, withdrawal, or trade involves blockchain transaction fees. These costs reduce the effective amount received.
```

Comment:
While this is true, they are very minimal on ICP usually. 
On ICP I would not call this "blockchain transaction fee" but rather "ledger transaction fee"